etc/kayobe/ansible/scripts/test_smartmon.py

```diff
@@ -0,0 +1,262 @@
+import glob
+import json
+import os
+import re
+import unittest
+
+from unittest.mock import patch, MagicMock
+
+from smartmon import (
+    parse_device_info,
+    parse_if_attributes,
+    main,
+    SMARTMON_ATTRS
+)
+
+def load_json_fixture(filename):
+    """
+    Load a JSON file from the 'drives' subfolder.
+    """
+    path = os.path.join(os.path.dirname(__file__), "drives", filename)
+    with open(path, "r", encoding="utf-8") as f:
+        return json.load(f)
+
+
+class TestSmartMon(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # Collect all *.json files from ./drives/
+        data_folder = os.path.join(os.path.dirname(__file__), "drives")
+        cls.fixture_files = glob.glob(os.path.join(data_folder, "*.json"))
+
+    def create_mock_device_from_json(self, device_info, if_attributes=None):
+        """
+        Given a 'device_info' dict and optional 'if_attributes', build
```

Comment:
Can you create an actual pySmart device object here from the file output?
Eg. https://github.com/truenas/py-SMART/blob/master/tests/test_device.py#L43

Reply:
So I gave this a shot but it turns out that it also requires a mock for smartctl and this made it a lot more complicated to just use a pySmart device. I think it makes more sense to mock it here

Reply:
Will do as a part 2